Pin scipy < 1.16.0 for `run` as well

mschilli87 opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/statsmodels-feedstock/issues/71.

<!--
Please add any other relevant info below:
-->

PR #72 added this workaround to the `host` environment, but not the `run` one.

This PR fixes this.

mschilli87 committed:
Pin scipy < 1.16.0 for `run` as well

Commits 9d9a8b31e9ca83d74b61246b894a4b7ca7ee9bbc added this workaround
to the `host` environment, but not the `run` one.

This commit fixes this.

Fixes https://github.com/conda-forge/statsmodels-feedstock/issues/71.

mschilli87 commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15991040528](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15991040528). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15981559796](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15981559796). Examine the logs at this URL for more detail.</sub>

bashtage commented:
I don't think the pin is actually needed for build since statsmodels code is not imported.  Probably only in run.

mschilli87 commented:
I can update this PR accordingly.

mschilli87 committed:
Drop scipy < 1.16.0 pin for `build`

This reverts the chnage of commit
9d9a8b31e9ca83d74b61246b894a4b7ca7ee9bbc as the pin is actually required
in the run environment only, which has been added by commits
59a89fad3931db8b200e41fcbad44648dab7d60d.

mschilli87 commented:
Done.

---
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15982060305](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15982060305). Examine the logs at this URL for more detail.</sub>

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -29,14 +29,14 @@ requirements:
     - numpy <3,>=1.22.3
     - python
-    - scipy !=1.9.2,>=1.8
+    - scipy !=1.9.2,>=1.8,<1.16.0
```
Let's flip the order here
```suggestion
    - scipy >=1.8,!=1.9.2,<1.16.0
```
For more details see https://github.com/conda-forge/conda-smithy/issues/2343

mschilli87 committed:
Re-order version limitations

Co-authored-by: h-vetinari <h.vetinari@gmx.com>

bashtage commented:
Thanks.
